website/addons/mendeley/views.py

```diff
@@ -0,0 +1,118 @@
+import httplib as http
+
+from flask import request
+
+from framework.exceptions import HTTPError
+from website.oauth.models import ExternalAccount
+from website.project.decorators import must_be_contributor_or_public
+from website.project.decorators import must_have_permission
+from website.project.decorators import must_not_be_registration
+from website.project.decorators import must_have_addon
+
+from .model import Mendeley
+
+
+@must_have_addon('mendeley', 'user')
+def list_mendeley_accounts_user(auth, user_addon):
+    return {
+        'accounts': [
+            {
+                'id': account._id,
```

Comment:
Since this logic occurs twice, how about writing a `serialize_account` method for it? This should even work for non-Mendeley accounts, given the generic structure you've added.